tlp-pd.in

```diff
@@ -0,0 +1,394 @@
+
+    # --- Interfaces org.freedesktop.UPower.PowerProfiles and net.hadess.PowerProfiles (legacy)
+
+    @dbus.service.method(
+        dbus_interface=INTERFACE_NAME,
+        in_signature="sss",
+        out_signature="u",
+        sender_keyword="sender",
+    )
+    @dbus.service.method(
+        dbus_interface=INTERFACE_NAME_LEGACY,
+        in_signature="sss",
+        out_signature="u",
+        sender_keyword="sender",
+    )
+    def HoldProfile(self, profile, reason, application_id, sender=None):
+        # Hold a profile until released
+        ### Q: do we need this resp. which client is actually using it?
+        ### TODO: extend to AVAILABLE_PROFILES?
+        if profile not in ["power-saver", "performance"]:
+            raise dbus.exceptions.DBusException(
+                f"Invalid profile: {profile}",
+                name="org.freedesktop.DBus.Error.InvalidArgs",
+            )
+
+        self._cookie_counter += 1
+        cookie = self._cookie_counter
+
+        self._holds[cookie] = {
+            "profile": profile,
+            "reason": reason,
+            "application_id": application_id,
+            "sender": sender,
+        }
+
+        syslog.syslog(
+            syslog.LOG_INFO,
+            f"Profile hold created: {application_id} holds {profile} (cookie: {cookie})",
+        )
+
+        # Update active profile based on holds
+        self._update_profile_from_holds()
+
+        return dbus.UInt32(cookie)
+
+    @dbus.service.method(
+        dbus_interface=INTERFACE_NAME, in_signature="u", out_signature=""
+    )
+    @dbus.service.method(
+        dbus_interface=INTERFACE_NAME_LEGACY, in_signature="u", out_signature=""
+    )
+    def ReleaseProfile(self, cookie):
+        # Release a profile hold
+        ### Q: do we need this resp. which client is actually using it?
+        cookie = int(cookie)
+        if cookie not in self._holds:
+            raise dbus.exceptions.DBusException(
+                f"Invalid cookie: {cookie}",
+                name="org.freedesktop.DBus.Error.InvalidArgs",
+            )
+
+        hold_info = self._holds.pop(cookie)
+        syslog.syslog(
+            syslog.LOG_INFO,
+            f"Profile hold released: {hold_info['application_id']} (cookie: {cookie})",
+        )
+
+        # Update active profile based on remaining holds
+        self._update_profile_from_holds()
+
+    @dbus.service.method(
+        dbus_interface=INTERFACE_NAME, in_signature="s", out_signature=""
+    )
+    @dbus.service.method(
+        dbus_interface=INTERFACE_NAME_LEGACY, in_signature="s", out_signature=""
+    )
+    def SyncProfile(self, profile):
+        # Callback for tlp after profile changes: update internal state only
+        if profile not in AVAILABLE_PROFILES:
+            raise dbus.exceptions.DBusException(
+                f"Invalid profile: {profile}",
+                name="org.freedesktop.DBus.Error.InvalidArgs",
+            )
+
+        if args.debug:
```

Comment:
Is there a good way to not have to write an if statement on each debug-level syslog call? I see python's logging library includes a [SysLogHandler](https://docs.python.org/3/library/logging.handlers.html#sysloghandler) and it also supports configuring minimum log levels on the logger object.